Mathlib/Logic/Function/Basic.lean

```diff
@@ -874,6 +874,44 @@ theorem sometimes_spec {p : Prop} {α} [Nonempty α] (P : α → Prop) (f : p 
 
 end Sometimes
 
+section Coequalizer
```

Comment:
As this is somewhat specialized, I would rather see that in a separate file like `Logic.Function.Coequalizer`. Otherwise, it looks good to me.